algorithms/src/Kokkos_Sort.hpp

```diff
@@ -146,8 +146,12 @@ class BinSort {
         Kokkos::is_view<SrcViewType>::value,
         Kokkos::View<typename SrcViewType::const_data_type,
                      typename SrcViewType::array_layout,
-                     typename SrcViewType::device_type,
-                     Kokkos::MemoryTraits<Kokkos::RandomAccess> >,
+                     typename SrcViewType::device_type
+#if not defined(KOKKOS_COMPILER_NVHPC)  // FIXME_NVHPC
```

Comment:
We typically don't use `not` in the code base.
```suggestion
#if !defined(KOKKOS_COMPILER_NVHPC)  // FIXME_NVHPC
```

Is that a compile-time error with NVHPC?

Reply:
ah good catch i will use !. 
No runtime error